daemon/openrazer_daemon/hardware/core.py

```diff
@@ -20,3 +20,20 @@ class RazerCore(__RazerDeviceBrightnessSuspend):
     METHODS = ['set_wave_effect', 'set_static_effect', 'set_spectrum_effect',
                'set_reactive_effect', 'trigger_reactive_effect', 'set_none_effect', 'set_breath_random_effect', 'set_breath_single_effect', 'set_breath_dual_effect',
                'set_custom_effect', 'set_key_row', 'get_device_type_core']
+
+
+class RazerCoreXChroma(__RazerDeviceBrightnessSuspend):
+    """
+    Class for the Razer Core
+    """
+    USB_VID = 0x1532
+    USB_PID = 0x0F1A
+    HAS_MATRIX = True
+    MATRIX_DIMS = [1, 16]
+
```

Comment:
@Ugzuzg Try setting `WAVE_DIRS = (0, 1)` here to see if it fixes wave direction. Otherwise I think some devices have some other value set in kernel driver but that's rather uncommon.

Reply:
Behaves in the same way.

https://user-images.githubusercontent.com/1078004/211918043-7cf346e2-3d54-4613-b31a-4fdb27c98b97.mp4

Reply:
Try using this codepath then please https://github.com/openrazer/openrazer/blob/master/driver/razeraccessory_driver.c#L661
And unset WAVE_DIRS again I guess

Reply:
@z3ntu, thanks, that works!

```diff
diff --git a/driver/razeraccessory_driver.c b/driver/razeraccessory_driver.c
index 56b7a1d..fcbce9d 100644
--- a/driver/razeraccessory_driver.c
+++ b/driver/razeraccessory_driver.c
@@ -649,7 +649,6 @@ static ssize_t razer_attr_write_matrix_effect_wave(struct device *dev, struct de
     case USB_DEVICE_ID_RAZER_MOUSE_BUNGEE_V3_CHROMA:
     case USB_DEVICE_ID_RAZER_BASE_STATION_V2_CHROMA:
     case USB_DEVICE_ID_RAZER_RAPTOR_27:
-    case USB_DEVICE_ID_RAZER_CORE_X_CHROMA:
         request = razer_chroma_extended_matrix_effect_wave(VARSTORE, ZERO_LED, direction);
         request.transaction_id.id = 0x1F;
         break;
@@ -658,6 +657,7 @@ static ssize_t razer_attr_write_matrix_effect_wave(struct device *dev, struct de
         // Must be in normal mode for hardware effects
         razer_set_device_mode(device->usb_dev, 0x00, 0x00);
         fallthrough;
+    case USB_DEVICE_ID_RAZER_CORE_X_CHROMA:
     case USB_DEVICE_ID_RAZER_THUNDERBOLT_4_DOCK_CHROMA:
         // Direction values are flipped compared to other devices
         direction ^= ((1<<0) | (1<<1));
```

Reply:
Nice!